src/Rules/Regexp/RegularExpressionQuotingRule.php

```diff
@@ -0,0 +1,182 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Rules\Regexp;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\BinaryOp\Concat;
+use PhpParser\Node\Expr\FuncCall;
+use PhpParser\Node\Name;
+use PHPStan\Analyser\Scope;
+use PHPStan\Rules\IdentifierRuleError;
+use PHPStan\Rules\Rule;
+use PHPStan\Rules\RuleErrorBuilder;
+use PHPStan\Type\Constant\ConstantStringType;
+use function array_merge;
+use function count;
+use function in_array;
+use function sprintf;
+use function strtolower;
+use function substr;
+
+/**
+ * @implements Rule<Node\Expr\FuncCall>
+ */
+class RegularExpressionQuotingRule implements Rule
+{
+
+	public function getNodeType(): string
+	{
+		return FuncCall::class;
+	}
+
+	public function processNode(Node $node, Scope $scope): array
+	{
+		if (!$node->name instanceof Node\Name) {
+			return [];
+		}
+
+		$functionName = strtolower((string) $node->name);
```

Comment:
You've forgot to do the usual hasFunction/getFunction dance because of functions in namespaces.